tests/framework/utils.py

```diff
@@ -30,3 +34,94 @@ def search_output_from_cmd(cmd: str,
 
     raise RuntimeError("Could not find '%s' in output for '%s'" %
                        (find_regex.pattern, cmd))
+
+
+def get_files_from(find_path: str,
+                   pattern: str,
+                   exclude_names: list = None):
+    """
+    Return a list of files from a given path, recursively.
+
+    :param find_path: path where to look for files
+    :param pattern: what pattern to apply to file names
+    :param exclude_names: list of folder names to exclude
+    :return: list of found files
+    """
+    found = []
```

Comment:
In order to avoid using `os` related functions mixed with `glob`, we could use an include/exclude approach based only on glob, something like:
```
    all = set((glob.glob(f"{find_path}/**/{pattern}",
                        recursive=True)))
    found = all
    for exclude in exclude_names:

        exclude_set = set(glob.glob(f"{find_path}/{exclude}/**/{pattern}",
                        recursive=True))
        found = found - exclude_set
    return found
```

Reply:
Is there a reason why we would like to avoid using `os` functions?

Reply:
No, I was saying that we can avoid **mixing** glob and os functions, but no strong reason against them. THis is more of a nit, anyways.